### PR TITLE
Getting procedures in the surface file

### PR DESCRIPTION
**Pull request by kwr3**
- adds two classes NexusProc and NexusProcs to handle specific data about the proc (NAME, PRIORITY, as well as the contents of the procedure)
- adds private method __load_procs that is called from the load() method when adding to memory in the NexusNetwork.py file
- test cases developed are in the test file

### CODE REVIEW

File: ResSimpy/Nexus/NexusNetwork.py [hunk 1]
```diff
@@ -91,6 +97,64 @@ def get_network_file(self, method_number: int = 1) -> NexusFile:
...
+            # boolean to determine if we are interested in a specific line or not
+            grab_line = False
+
+            # the contents of a specific procedure as a list
+            proc_contents: list[str] = []
+
+            # initialize the time to the start date of the model, name, and priority
+            time = self.__model.start_date
+            name_ = None
+            priority_ = None
```
Comment (jonnyh404) [inline ResSimpy/Nexus/NexusNetwork.py:127]: Any specific reasoning to the underscore on these variable names?

File: tests/Nexus/nexus_simulator/test_nexus_network_procedures.py
Comment (jonnyh404) [inline tests/Nexus/nexus_simulator/test_nexus_network_procedures.py:1]: Some of these tests duplicate a lot of code, these could be parameterized to save duplication.

**Review by jonnyh404 — APPROVED** [review]
Good set of changes to bring in reading the PROCS. Please take a look at comments and decide what could be refactored with the tests. I don't think they need to all be parameterized.

File: tests/Nexus/nexus_simulator/test_nexus_network_procedures.py
Comment (kwr3) [reply tests/Nexus/nexus_simulator/test_nexus_network_procedures.py:1]: will set up some time to look at parameterized testing with you. would be great practice

File: ResSimpy/Nexus/NexusNetwork.py (hunk 1 above)
Comment (kwr3) [reply ResSimpy/Nexus/NexusNetwork.py:127]: My logic was to differentiate them when I create an object i.e.
proc_obj = NexusProc(contents=proc_contents, date=time, name=name_, priority=priority_)

happy to remove if unnecessary

**Pull request merged**